twake/backend/node/src/services/messages/services/views/service.ts

```diff
@@ -222,6 +222,7 @@ export class ViewsService implements MessageViewsServiceAPI {
         ...(options.companyId ? { $in: [["company_id", [options.companyId]]] } : {}),
         ...(options.workspaceId ? { $in: [["workspace_id", [options.workspaceId]]] } : {}),
         ...(options.channelId ? { $in: [["channel_id", [options.channelId]]] } : {}),
+        // TODO: implement boolean for hasFiles
```

Comment:
You can already do it:

```
async search(
    pagination: Pagination,
    options?: SearchMessageOptions,
    context?: ExecutionContext,
  ): Promise<ListResult<Message>> {
    return await this.searchRepository.search(
      {
      //Here:
        has_file: myBoolean
      },
      {
        pagination,
        ...(options.companyId ? { $in: [["company_id", [options.companyId]]] } : {}),
        ...(options.workspaceId ? { $in: [["workspace_id", [options.workspaceId]]] } : {}),
        ...(options.channelId ? { $in: [["channel_id", [options.channelId]]] } : {}),
```

Reply:
Same place for the `sender`

Reply:
Couldn't handle with has_files actually, need your help.